### PR TITLE
support scala3 in avroparquet

### PR DESCRIPTION
**Pull request by pjfanning**
part of https://github.com/apache/incubator-pekko-connectors/issues/126

### CODE REVIEW

**Comment by pjfanning** [comment]
@mdedetrich @He-Pin @nvollmar would any of you time to review this? It is the final easy Scala3 transition. The other harder ones have issues raised for them - https://github.com/apache/incubator-pekko-connectors/issues?q=is%3Aissue+is%3Aopen+label%3Ascala3

File: project/Dependencies.scala [hunk 1]
```diff
@@ -167,13 +167,16 @@ object Dependencies {
       "com.google.jimfs" % "jimfs" % "1.2" % Test // ApacheV2
     ))
 
+  val avro4sVersion: Def.Initialize[String] = Def.setting {
+    if (scalaVersion.value.startsWith("2")) "4.1.1" else "5.0.4"
```
Comment (He-Pin) [inline project/Dependencies.scala:171]: I see.

File: avroparquet/src/test/scala/docs/scaladsl/AbstractAvroParquetBase.scala [hunk 2]
```diff
@@ -42,13 +34,13 @@ trait AbstractAvroParquet extends BeforeAndAfterAll {
...
 
-  val format: RecordFormat[Document] = RecordFormat[Document]
-
   val folder: String = "./" + Random.alphanumeric.take(8).mkString("")
 
   val genFinalFile: Gen[String] = for {
     fileName <- Gen.alphaLowerStr
-  } yield { folder + "/" + fileName + ".parquet" }
+  } yield {
+    folder + "/" + fileName + ".parquet"
```
Comment (He-Pin) [inline avroparquet/src/test/scala/docs/scaladsl/AbstractAvroParquetBase.scala:42]: use File.seperator?

Comment (pjfanning) [reply avroparquet/src/test/scala/docs/scaladsl/AbstractAvroParquetBase.scala:42]: I'm not sure how the code got reformatted. Possibly my IDE - possibly scalafmt. If this is an issue I can try to change the formatting of this line back.

File: build.sbt [hunk 3]
```diff
@@ -111,10 +111,26 @@ TaskKey[Unit]("verifyCodeFmt") := {
   }
 }
 
+val scalaReleaseSeparateSource: Def.SettingsDefinition = Compile / unmanagedSourceDirectories ++= {
```
Comment (mdedetrich) [inline build.sbt:114]: Why is this necessary, this functionality is automatically provided by sbt, i.e. sbt will automatically handle `scala-2`/`scala-3` folders in the way this code seems to be describing.

You only need custom handling if you want to do something like `scala-2.13+` like in pekko core

File: project/Dependencies.scala (hunk 1 above)
[the remark this answers is not in the document]
Comment (pjfanning) [reply project/Dependencies.scala]: done

File: build.sbt (hunk 3 above)
Comment (pjfanning) [reply build.sbt]: I wasn't aware that scala-2, scala-3 worked automatically, I've removed explicit setting now.
